### PR TITLE
brainbuilder@0.12.0

### PR DESCRIPTION
**Pull request by asanin-epfl**
I need help here with py-pandas@1.0.3. It does not compile.

@wizmer I'm just going to put it here https://github.com/BlueBrain/morph-tool/pull/13#discussion_r425001873

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pandas/package.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ class PyPandas(PythonPackage):
         'pandas.api.extensions'
     ]
 
+    version('1.0.3', sha256='32f42e322fb903d0e189a4c10b75ba70d90958cc4f66a1781ed027f1a1d14586')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-pandas/package.py]: What feature is needed from this?

File: var/spack/repos/builtin/packages/brainbuilder/package.py [hunk 2]
```diff
@@ -37,4 +38,6 @@ class Brainbuilder(PythonPackage):
     depends_on('py-tqdm@4.0:', type='run')
 
     depends_on('py-bluepy@0.12.5:', type='run')
-    depends_on('py-voxcell@2.6.2:', type='run')
+    depends_on('py-voxcell@:2.6.9', type='run', when='@:0.11.7')
+    depends_on('py-voxcell@2.7:', type='run', when='@0.12:')
+    depends_on('py-morph-tool@0.2.3:', type='run', when='@0.11.9:')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/brainbuilder/package.py:43]:
```
py-morph-tool requires py-neurom version 1.4.15:, but spec asked for mut_morphio
```
Something is messed up.

File: var/spack/repos/builtin/packages/py-pandas/package.py (hunk 1 above)
Comment (wizmer) [reply var/spack/repos/builtin/packages/py-pandas/package.py]: It's not a feature but users are encouraged to move away from 0.25 as it was a transitional version. 0.25 and 1.0.3 are not fully compatible either. Also pandas 1.+ is the first Pandas that can be considered as stable.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-pandas/package.py]: In that case, I'd rather consider backporting (i.e. copying over) the package from upstream Spack. I was just wondering if there's a hard reason to switch.

We're also due for a deployment update soon™, that would probably bring an update to Pandas anyways. If you want it now and copy the official package over, then that will make merging upstream easier.

Comment (wizmer) [reply var/spack/repos/builtin/packages/py-pandas/package.py]: Yes, that was my original idea as well. I will take care of it.

File: var/spack/repos/builtin/packages/brainbuilder/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/brainbuilder/package.py:43]: See the last item of the [FAQ](https://github.com/BlueBrain/spack/blob/develop/FAQ.md). If you don't do the `spacktivate`, but use your local Spack with the branch on the pull request, you can roll back the dependencies one by one and see when it breaks.

**Comment by tomdele** [comment]
Pandas 1.0.3 is python 3 only. 
I though we wanted to keep brainbuilder python2 compatible for a while ?

**Comment by matz-e** [comment]
> Pandas 1.0.3 is python 3 only.
> I though we wanted to keep brainbuilder python2 compatible for a while ?

Our deployment is Python3 only already. Note that this does not impede your ability to lower the requirements in `setup.py` or on PyPI.

**Comment by tomdele** [comment]
@matz-e  Yes indeed... I got confused. Thanks for the reminder

**Review by matz-e — CHANGES_REQUESTED** [review]
`py-plotly-helper` contains this:/
```
depends_on('py-neurom@mut_morphio+plotly', type='run')
```
which breaks here. Is that `@mut_morphio` really necessary for a *helper*?

**Comment by asanin-epfl** [comment]
@wizmer can you please clarify on `py-plotly-helper` and its dependency to `py-neurom@mut_morphio+plotly`?
Should we update dependency in `py-morph-tool/package.py` to `depends_on('py-neurom@1.4.15:+mut_morphio', type='run', when='@0.1.14:')` ?

**Comment by matz-e** [comment]
> @wizmer can you please clarify on `py-plotly-helper` and its dependency to `py-neurom@mut_morphio+plotly`?
> Should we update dependency in `py-morph-tool/package.py` to `depends_on('py-neurom@1.4.15:+mut_morphio', type='run', when='@0.1.14:')` ?

There's no `+mut_morphio`, it's a version, so `@mut_morphio`. Really I would not expect a `plotly` helper to be so version dependent.

**Comment by wizmer** [comment]
Plotly-helper does not require @mut_morphio so it can be removed if needed.

**Comment by asanin-epfl** [comment]
ok, I will remove it

**Comment by asanin-epfl** [comment]
Finally Test Build passed

**Comment by asanin-epfl** [comment]
Should I retest this? travis is hanged for some reason

**Comment by matz-e** [comment]
> Should I retest this? travis is hanged for some reason

Grr… Travis seems a little sketchy over the last ~week. I'll have a look. In the web interface, it actually failed. Just retriggered Travis…

**Review by matz-e — DISMISSED** [review]
Thanks, looks good now. Just need to satisfy Travis…

**Comment by asanin-epfl** [comment]
Hooray!

**Review by tomdele — APPROVED** [review]

**Pull request merged**